src/main/scala/calc/Compiler.scala

```diff
@@ -74,8 +99,73 @@ object Compiler {
       case Literal(value) =>
         irt.DoubleLiteral(value)
 
-      case _ =>
-        throw new Exception(
+      case BinaryOp(op, lhs, rhs) => {
+        val jsop = op match {
+          case "+" => irt.BinaryOp.Double_+
+          case "-" => irt.BinaryOp.Double_-
+          case "*" => irt.BinaryOp.Double_*
+          case "/" => irt.BinaryOp.Double_/
+        }
+        irt.BinaryOp(jsop, irt.Unbox(compileExpr(lhs),'D'), irt.Unbox(compileExpr(rhs), 'D'))
+      }
+
+      case Let(ident, value, body) => {
+        var jsV:irt.Tree = if (value.isInstanceOf[Closure]){
+          compileClosure(value.asInstanceOf[Closure], ident.name)
```

Comment:
Oh so you basically did interpret `let` as a `letrec`.
Could make sense, but translation into the IR will be very hard. It is not asked that you support recursion.